src/std/string.md

```diff
@@ -27,8 +27,14 @@ fn main() {
 
 <details>
 
+* `new` heap allocated buffer. `StringL::with capacity` is used when you know how much you want to push on
 * `len` returns the size of the `String` in bytes, not its length in characters.
 * `chars` returns an iterator over the actual characters.
-* `String` implements `Deref<Target = str>` which transparently gives it access to `str`'s methods.
+*  When people refer to strings they could either be talking about &str or String. 
+* Implementing Deref (trait), gives the compiler ability to take a value of any type, call the deref method, and know how to dereference it  
+    * `String` implements `Deref<Target = str>` which transparently gives it access to `str`'s methods.
+    * Write and compare `let s3 = s1.deref();` and  `let s3 = &*s1`;.
+* String isimplemented as a wrapper around a vector of bytes, many of the operations you see supported on vectors are also supported on String, but with some extra guarantees.
+* Show the danger of indexing Strings by (1) adding a unicode character to an above string (Ex. `Hβello`) and slicing it at various locations and (2) unwraping the characters `.chars.nth(i).unwrap()` where `i` is in-bound and out-of-bounds given the number of chars. 
```

Comment:
I think this is outdated now: we already have a Unicode flag (🇨🇭) above.

Reply:
I updated the text. Let me know what you think.